src/main/java/seedu/module/model/module/TrackedModule.java

```diff
@@ -66,13 +72,39 @@ public void markDeadlineTaskAsDone(int taskListNum) {
         deadlineList.markDeadlineTaskAsDone(taskListNum);
     }
 
+    public void markAllDone() {
+        deadlineList.markAllDone();
+    }
+
+    public void markDeadlineTaskAsUndone(int taskListNum) throws CommandException {
+        deadlineList.markDeadlineTaskAsUndone(taskListNum);
+    }
+
     public void markDeadlineTaskAsInProgress(int taskListNum) {
         deadlineList.markDeadlineTaskAsInProgress(taskListNum);
     }
 
     public void deleteDeadlineTask(int taskListNum) {
         deadlineList.deleteDeadlineTask(taskListNum);
     }
+    public void deleteAllDeadlineTasks() {
+        deadlineList.deleteAllDeadlineTasks();
+    }
+
+    /**
+     * Returns true if there exists a deadline with the same description, time and tag
+     * @param deadline
+     * @return
+     */
+    public boolean hasDeadline(Deadline deadline) {
+        for (Deadline d : deadlineList.getDeadlineList()) {
+            if (deadline.getDescription().equals(d.getDescription()) && deadline.getTime().equals(d.getTime())
+                    && deadline.getTag().equals(d.getTag())) {
```

Comment:
Can this be simplified to `d.equals(deadline)`

Reply:
I think Ill keep this one as tasks with different priority still reflects as duplicate task.